ecmascript/transforms/react/src/jsx/mod.rs

```diff
@@ -1130,41 +1130,34 @@ fn to_prop_name(n: JSXAttrName) -> PropName {
 
 #[inline]
 fn jsx_text_to_str(t: JsWord) -> JsWord {
-    static SPACE_NL_START: Lazy<Regex> =
-        Lazy::new(|| Regex::new("^[\t'\n\x0C\r ]*\n[\t'\n\x0C\r ]*").unwrap());
-    static SPACE_NL_END: Lazy<Regex> =
-        Lazy::new(|| Regex::new("[\t'\n\x0C\r ]*\n[\t'\n\x0C\r ]*$").unwrap());
-
-    if t == *" " {
-        return t;
-    }
-
-    if !t.contains(' ') && !t.contains('\n') {
-        return t;
-    }
-
-    let s = SPACE_NL_START.replace_all(&t, "");
-    let s = SPACE_NL_END.replace_all(&s, "");
-    let need_leading_space = s.starts_with(' ');
-    let need_trailing_space = s.ends_with(' ');
-
-    let mut buf = String::from(if need_leading_space { " " } else { "" });
-
-    for (last, s) in s
-        .split(|c: char| c != '\u{a0}' && c.is_ascii_whitespace())
-        .filter(|s| !s.is_empty())
-        .identify_last()
-    {
-        buf.push_str(s);
-        if !last {
-            buf.push(' ');
+    static SPACE_START: Lazy<Regex> = Lazy::new(|| Regex::new("^[ ]+").unwrap());
+    static SPACE_END: Lazy<Regex> = Lazy::new(|| Regex::new("[ ]+$").unwrap());
+    let mut buf = String::from("");
```

Comment:
Can you this change to `String::new()`?

Reply:
of course.
btw, I looked into some failed test case, noticing their output is different with the output which I got with tsc playground, should i edit them? most of them added in #2456

Reply:
Yes, those tests exist to show difference of output of lots of inputs.
You can run `UPDATE=1 cargo test` from the repository root to update them.
Note that you may need to run it multiple time, until it exists normally.